src/concepts/kafka-proxies/grpc-proxy.md

```diff
@@ -1,297 +1,61 @@
-```
-
-- Routes for individual service methods
-
-```yaml
-routes:
-  - when:
-      - method: routeguide.RouteGuide/GetFeature
+```protobuf:no-line-numbers
+rpc SayHello(HelloRequest) returns (HelloResponse);
 ```
 
-## Step 2: Handling message routing onto Kafka
-
-This maps the proto service method's request and response messages directly to Kafka topics defined by the `topic` and `reply-to` attributes respectively. The messages are linked by the `zilla:correlation-id` header for individual calls into the gRPC service. Read more about it and how the `idempotency-key` enables safe message replays in the [grpc-kafka](../../reference/config/bindings/binding-grpc-kafka.md#produce-capability) reference section.
-
-### Message routing for RPC request and response types
-
-Let's look at some common service definitions and how Zilla can route their messages.
-
-#### Simple RPC
+A single message is sent and will wait for the correlated response message and return it back to the caller. The request and response topics both have one message with the method payloads and one end-of-stream message.
 
-For the `GetFeature` method, the `Point` request and `Feature` response create the stream of information Kafka can help manage.
+### Server-side streaming RPC
 
-```yaml
-routes:
-  - when:
-      - method: routeguide.RouteGuide/GetFeature
-    with:
-      capability: produce
-      topic: request-topic
-      reply-to: response-topic
+```protobuf:no-line-numbers
+rpc LotsOfReplies(HelloRequest) returns (stream HelloResponse);
 ```
 
-#### Server-side streaming RPC
+A single message is sent with a returned stream back to the caller. The correlated messages produced on the reply-to topic will be sent for the client to read until there are no more messages, and the stream will close. The request topic has one message with the method payloads and one end-of-stream message. The response topic has one or many messages with the method payloads and one end-of-stream message.
 
-For the `ListFeatures` method, we can have a dedicated stream of `Feature` messages being recorded on a topic before being sent to the client. Kafka allows the server to produce all of the responses without considering the connection to the client.
+### Client-side streaming RPC
 
-```yaml
-routes:
-  - when:
-      - method: routeguide.RouteGuide/ListFeatures
-    with:
-      capability: produce
-      topic: request-topic
-      reply-to: feature-stream-topic
+```protobuf:no-line-numbers
+rpc LotsOfGreetings(stream HelloRequest) returns (HelloResponse);
 ```
 
-#### Client-side streaming RPC
+The client sends a stream, producing all the messages on a topic. When the client finishes writing to the stream, it will wait for the correlated response message and return it back to the caller. The request topic has one or many messages with the method payloads and one end-of-stream message. The response topic has one message with the method payloads and one end-of-stream message.
```

Comment:
It's possible that the correlated response message can be sent before the request stream has completed.